### PR TITLE
First pass on compiler tests for full blocks

### PR DESCRIPTION
**Pull request by guillep**
Integrating full blocks for real requires that we do a pass on the compiler tests.
The compiler tests are right now using the default compiler, thus they cannot test the full blocks.

This PR introduces:
 - more compiler tests
 - refactored compiler tests to configure the bytecode backend and optimizations
 - extending SUnit and Parametrized tests to allow combination/multiplication of test matrixes

### CODE REVIEW

**Comment by guillep** [comment]
In the middle, I've found a mismatch between the optimized and non-optimized versions of whileTrue/False

```
b1 := [ 4 < 3 ].
b2 := [  ].
b1 whileTrue: b2.
  => "the receiver block"

[ 4 < 3 ] whileTrue: [  ].
  => nil
```

Which can be also seen in
```
OpalCompiler new
	evaluate: '[ 4 < 3 ] whileTrue: [  ]'.
  => "nil"
	
OpalCompiler new
	options: #( -optionInlineWhile );
	evaluate: '[ 4 < 3 ] whileTrue: [  ]'.
  => "the receiver block"
```

I propose that the `whileTrue:` and `whileFalse:` methods do return nil to behave as the optimized version.

File: src/Kernel/Process.class.st [hunk 1]
```diff
@@ -39,14 +39,6 @@ Class {
...
 }
 
-{ #category : #'process specific' }
-Process class >> psKeysSema [
-	"Isolate handling of class variable"
-
-	PSKeysSema ifNil: [ PSKeysSema := Semaphore forMutualExclusion ].
-	^PSKeysSema
-]
-
```
Comment (guillep) [inline src/Kernel/Process.class.st:49]: This is super strange, I don't understand why this method got moved down...
Actually, my commit is sorting it. So somebody commited non-sorted methods?

Comment (Ducasse) [reply src/Kernel/Process.class.st:49]: I saw this code in recent PRs but I do not remember which one.

Comment (guillep) [reply src/Kernel/Process.class.st:49]: https://github.com/pharo-project/pharo/pull/4106/files

**Comment by Ducasse** [comment]
Guille do we wait that all the tests are green? Because there some broken tests.

**Comment by guillep** [comment]
Yes, I have some little stuff to do here yet. Please do not integrate until tests are green :)

**Comment by Ducasse** [comment]
So we integrate it?

**Comment by guillep** [comment]
I'd say yes. There is a failure in the build but due to the following test that fails from time to time.
```
0xffe8fc30 M [] in FLContextSerializationTest>testDoIt 0x8deb498: a(n) FLContextSerializationTest
```

**Pull request merged**